arkane/ess/adapter.py

```diff
@@ -166,19 +180,3 @@ def get_symmetry_properties(self):
             return optical_isomers, symmetry, pg.point_group
         finally:
             shutil.rmtree(scr_dir)
-
-    @abstractmethod
-    def get_D1_diagnostic(self):
-        """
-        This method returns the D1 diagnostic for certain quantum jobs
-        Should be implemented by a subclass.
-        """
-        pass
-
-    @abstractmethod
-    def get_T1_diagnostic(self):
```

Comment:
Ah my guess is it would throw an error since in that case neither the child nor parent class implements it. If I make get_D1_diagnostic and get_T1_diagnostic regular methods, rather than abstract methods, in ESSAdapter, that seems to solve the problem since it correctly raises the NotImplementedError from the subclass, such as `NotImplementedError: The get_D1_diagnostic method is not implemented for Gaussian Logs`.

Reply:
I see. Would you like to make them regular methods?

Reply:
Yes doing it now! I tested the code below in a separate terminal so it seems to work. Just wanted to check with you if I was missing anything before committing and squashing this change :) 

```
>>> object = Log('/Users/kevin/Documents/Green_Lab/RMG/RMG-Py/arkane/data/Benzyl/BenzylFreq.log')
>>> object.get_number_of_atoms()
14
>>> object.get_D1_diagnostic()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kevin/Documents/Green_Lab/RMG/RMG-Py/arkane/ess/factory.py", line 231, in get_D1_diagnostic
    return self.ess.get_D1_diagnostic()
  File "/Users/kevin/Documents/Green_Lab/RMG/RMG-Py/arkane/ess/gaussian.py", line 514, in get_D1_diagnostic
    raise NotImplementedError('The get_D1_diagnostic method is not implemented for Gaussian Logs')
NotImplementedError: The get_D1_diagnostic method is not implemented for Gaussian Logs
>>> object.get_T1_diagnostic()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/kevin/Documents/Green_Lab/RMG/RMG-Py/arkane/ess/factory.py", line 237, in get_T1_diagnostic
    return self.ess.get_T1_diagnostic()
  File "/Users/kevin/Documents/Green_Lab/RMG/RMG-Py/arkane/ess/gaussian.py", line 518, in get_T1_diagnostic
    raise NotImplementedError('The get_T1_diagnostic method is not implemented for Gaussian Logs')
NotImplementedError: The get_T1_diagnostic method is not implemented for Gaussian Logs
```

Reply:
So `get_X1_diagnostic` is a method in each adapter that just raises a NonImplemented Error?
But what if someone adds a new Adapter and leaves this non-required method?

Reply:
Some adapters implement the method. For example, OrcaLog implements a `get_T1_diagnostic` method and MolproLog implements both a `get_T1_diagonistic` and `get_D1_diagnostic`. GaussianLog implements neither so it correctly raises the error, as it did previously when GaussianLog inherited from Log

Reply:
I updated them to be regular methods in ESSAdapter so it is optional if the adapter child classes want to override them with an actual method, rather than just overriding them with an error as they did previously